vllm/model_executor/layers/quantization/compressed_tensors/compressed_tensors.py

```diff
@@ -0,0 +1,155 @@
+from typing import Any, Dict, List, Optional
+
+import torch
+
+from vllm.model_executor.layers.linear import LinearMethodBase
+from vllm.model_executor.layers.quantization.base_config import (  # noqa: E501
+    QuantizationConfig)
+
+from vllm.model_executor.layers.quantization.compressed_tensors.schemes import (
+    CompressedTensorsW8A8StaticTensor, CompressedTensorsUnquantized,
+    CompressedTensorsScheme)
+from vllm.model_executor.utils import set_weight_attrs
+
+
+class CompressedTensorsConfig(QuantizationConfig):
+
+    def __init__(self, layer_quant_details: Dict[str, Any], ignore: List[str],
+                 fake_quant: bool):
+        self.fake_quant = fake_quant
+        self.ignore = ignore
+        self.layer_quant_details = layer_quant_details
+
+    def get_linear_method(self) -> "CompressedTensorsLinearMethod":
+        return CompressedTensorsLinearMethod(self)
+
+    def get_scaled_act_names(self) -> List[str]:
+        return []
+
+    def get_supported_act_dtypes(cls) -> List[torch.dtype]:
+        return [torch.float32, torch.int8]
+
+    # Need to figure it out
+    def get_min_capability(self) -> int:
+        return 60
+
+    def get_name(self) -> str:
+        return "compressed_tensors"
+
+    @classmethod
+    def from_config(cls, config: Dict[str, Any]) -> "CompressedTensorsConfig":
+        layer_quant_details: Dict[str:Any] = dict()
+        ignore = config.get("ignore")
+        fake_quant = config.get("format") == "fakequant"
+
+        for key, quant_config in config["config_groups"].items():
+            targets = quant_config.get("targets")
+            for target in targets:
+                layer_quant_details[target] = {}
+                layer_quant_details[target]["weight"] = quant_config.get(
+                    "weights")
+                layer_quant_details[target]["input"] = quant_config.get(
+                    "input_activations")
+
+        return cls(layer_quant_details=layer_quant_details,
+                   ignore=ignore,
+                   fake_quant=fake_quant)
+
+    @classmethod
+    def get_config_filenames(cls) -> List[str]:
+        return ["config.json"]
+
+    def _get_schema(self, weight_quant: Dict, input_quant: Dict):
```

Comment:
agree if we can clean this up it would be good

Reply:
yup doing this now

Reply:
PR to clean-up some of the parsing: https://github.com/neuralmagic/compressed-tensors/pull/44